src/geventhttpclient/useragent.py

```diff
@@ -437,8 +437,10 @@ def urlopen(self, url, method='GET', response_codes=valid_response_codes,
                 payload = kwargs
             elif isinstance(payload, dict):
                 payload.update(kwargs)
-
-        req = self._make_request(url, method=method, headers=headers, payload=payload, params=params, **kwargs)
+            files = kwargs.get("files", None)
```

Comment:
wouldn't it be better to add `files` to the named argument list of `urlopen`?

right now, if `payload` is `False`, and one have `files=` in the `urlopen` call, it would be present in both `payload` and used as an for the `files` argument to the `_make_request` call.

Reply:
fixing

Reply:
The weird thing is that it doesnt work if I make it like this either:

    def urlopen(self, url, method='GET', response_codes=valid_response_codes,
                headers=None, payload=None, to_string=False, debug_stream=None, params=None, **kwargs):
        """ Open an URL, do retries and redirects and verify the status code
        """
        # POST or GET parameters can be passed in **kwargs
        if kwargs:
            files = kwargs.pop("files", None)
            if not payload:
                payload = kwargs
            elif isinstance(payload, dict):
                payload.update(kwargs)
        else:
            files = None
        req = self._make_request(url, method=method, headers=headers, payload=payload, params=params, files=files)